File: shoring/compat/src/test/java/com/panopset/compat/test/FileProcessorTest.java
package com.panopset.compat.test;

import java.io.File;
import java.io.IOException;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.panopset.compat.FileProcessor;
import com.panopset.compat.Fileop;
import com.panopset.compat.LineFilter;
import com.panopset.compat.Stringop;

public class FileProcessorTest {

  final File file = FileopTest.tempFile;

  @Test
  void test() throws IOException {
    Fileop.delete(file);
    Fileop.write(Stringop.FOO, file);


    FileProcessor fp = new FileProcessor(file).withLineFilter(new LineFilter() {

      @Override
      public String filter(String str) {
        return Stringop.BAR;
      }

    });
    Assertions.assertTrue(fp.exec());
    String result = Fileop.readTextFile(file);
    Assertions.assertEquals(Stringop.appendEol(Stringop.BAR), result);
    FileopTest.cleanup();
  }

  @Test
  void testNullResultFilter() throws IOException {
    Fileop.delete(file);
    Fileop.write(Stringop.FOO, file);
    FileProcessor fp = new FileProcessor(file).withLineFilter(new LineFilter() {

      @Override
      public String filter(String str) {
        return null;
      }

    });
    Assertions.assertTrue(fp.exec());
    FileopTest.cleanup();
  }
}

File: code/shoring/compat/src/test/kotlin/com/panopset/compat/test/FileProcessorTest.kt
package com.panopset.compat.test

import com.panopset.compat.*
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File
import java.io.IOException

class FileProcessorTest {
    val file: File = tempFile
    @Test
    fun test() {
        Fileop.delete(file)
        Fileop.write(Stringop.FOO, file)
        val fp = FileProcessor(file).withLineFilter(MyLineFilter())
        fp.exec()
        val result = Fileop.readTextFile(file)
        Assertions.assertEquals(Stringop.BAR, result)
        FileopTest.cleanup()
    }
}

class MyLineFilter: ByLineFilter {
    override fun filter(str: String): FilteredString {
        return FilteredString(Stringop.BAR)
    }
}
